### PR TITLE
write_inp_section changes - don't change the Polygons header

### LINKED ISSUE #181: Polygons header overwrite bug
Below is an explanation and steps to reproduce the Polygons header overwriting bug. 

The event occurs when using the "replace_inp_section" function. 
The code I am executing: 
```python
def _add_coords(self, subcatchment_id):
    exist = [
        (self.model.inp.polygons["X"][-1], self.model.inp.polygons["Y"][-1]),
        (self.model.inp.polygons["X"][-2], self.model.inp.polygons["Y"][-2]),
        (self.model.inp.polygons["X"][-3], self.model.inp.polygons["Y"][-3]),
        (self.model.inp.polygons["X"][-4], self.model.inp.polygons["Y"][-4]),
    ]
    coords = pd.DataFrame(
        data={
            "X": [exist[0][0], exist[1][0], exist[2][0], exist[3][0]],
            "Y": [exist[0][1] - 5, exist[1][1] - 5, exist[2][1] - 5, exist[3][1] - 5],
        },
        index=[subcatchment_id for _ in range(4)]
    )
    coords.index.names = ['Name']
    self.model.inp.polygons = pd.concat([self.model.inp.polygons, coords])
    replace_inp_section(self.model.inp.path, "[Polygons]", self.model.inp.polygons)
```
I see that SWMM writes literally "Polygons" in the *inp file, swmmio modifies this name by writing "POLYGONS".

Here is the section I have in the file before executing the code
```python
[Polygons]
;;Subcatchment   X-Coord            Y-Coord           
;;-------------- ------------------ ------------------
S1               777180.000         592590.000        
S1               777180.000         592585.000        
S1               777175.000         592585.000        
S1               777175.000         592590.000        
```
This is what the section looks like after the first execution of "replace_inp_section"
```python
[POLYGONS]
;;   X          Y         
S1   777180.0   592590.0
S1   777180.0   592585.0
S1   777175.0   592585.0
S1   777175.0   592590.0
S2   777175.0   592585.0
S2   777175.0   592580.0
S2   777180.0   592580.0
S2   777180.0   592585.0
```
This is what the sections looks like after the second execution of "replace_inp_section"
```python
[POLYGONS]
;;   X          Y         
S1   777180.0   592590.0
S1   777180.0   592585.0
S1   777175.0   592585.0
S1   777175.0   592590.0
S2   777175.0   592585.0
S2   777175.0   592580.0
S2   777180.0   592580.0
S2   777180.0   592585.0

[POLYGONS]
;;   X          Y         
S1   777180.0   592590.0
S1   777180.0   592585.0
S1   777175.0   592585.0
S1   777175.0   592590.0
S2   777175.0   592585.0
S2   777175.0   592580.0
S2   777180.0   592580.0
S2   777180.0   592585.0
S3   777180.0   592580.0
S3   777180.0   592575.0
S3   777175.0   592575.0
S3   777175.0   592580.0
```
In the code I execute I always use "[Polygons]", after the second execution of "replace_inp_section" the sections are duplicated. 
The problem doesn't occur when I change the header to "[POLYGONS]" before executing the code, and in my function I also use "[POLYGONS]"

Steps to solve the problem:
- [x] check when the header is overwritten
- [x] handle the correct header overwrite

### PR DESCRIPTION
**Pull request by BuczynskiRafal**
Closes #181

### CODE REVIEW

**Comment by BuczynskiRafal** [comment]
@aerispaha I'm not sure how to test the solution. Can I find a method in swmmio that reads the header in the inp file to compare it with the header after doing replace_inp_section?

**Comment by bemcdonnell** [comment]
@BuczynskiRafal, I'm guessing you're adding this feature because of some kind of text version control issue.   I think the best way to test it is to add a second INP file with `[POLYGONS]` and check the outputs from that. You can use a simple file parser to assert that the text found is either your `[Polygons]` or `[POLYGONS]`

**Comment by BuczynskiRafal** [comment]
@aerispaha @bemcdonnell Please review my PR. Thank you.

File: swmmio/tests/data/__init__.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 MODEL_EX_1_PARALLEL_LOOP = os.path.join(DATA_PATH, 'Example1_parallel_loop.inp')
 MODEL_INFILTRAION_PARSE_FAILURE = os.path.join(DATA_PATH, 'model-with-infiltration-parse-failure.inp')
 MODEL_EXTCNTRLMODEL = os.path.join(DATA_PATH, 'SWMMExtCntrlModel.inp')
+MODEL_POLYGONS = os.path.join(DATA_PATH, 'example_polygons.inp')
```
Comment (aerispaha) [inline swmmio/tests/data/__init__.py:34]: I don't think we need to add this new inp file to test that this bug fix works properly. Any existing test inp file in the repo with a Polygons section will do.

File: swmmio/tests/data/model_full_features_network_xy.inp [hunk 2]
```diff
@@ -312,31 +312,30 @@ C2.1             2747634.497        1117846.256
...
-S3               2748511.859        1117807.464       
-S4               2746913.127        1118252.931       
-S4               2746942.825        1118500.413       
-S4               2747192.781        1118468.241       
-S4               2747168.033        1118599.406       
-S4               2747088.839        1118755.319       
-S4               2746741.127        1118938.456       
-S4               2746527.056        1118812.240       
-S4               2746551.804        1118480.615       
+;;   X             Y
```
Comment (aerispaha) [inline swmmio/tests/data/model_full_features_network_xy.inp:315]: Are these changes necessary for anything? If not, let's remove these changes from the PR

File: swmmio/tests/test_version_control.py [hunk 3]
```diff
@@ -220,3 +221,29 @@ def test_modify_inp_sections():
         sects1 = get_inp_sections_details(m1.inp.path)
         sects2 = get_inp_sections_details(m2.inp.path)
         assert all([x in sects1 for x in sects2])
+
+
+def test_replace_inp_section_overwrite_polygons_header():
```
Comment (aerispaha) [inline swmmio/tests/test_version_control.py:226]: This tests that the bug fix works properly which is great! 

But I'd prefer store any test artifacts in a temporary file location, this way if the test fails, it won't leave behind files in the project directory. How do you feel about refactoring this unit test like this: 

```python
def test_replace_inp_section_overwrite_polygons_header():
    model = swmmio.Model(MODEL_A_PATH)

    with tempfile.TemporaryDirectory() as tempdir:

        # create a temporary inp file path
        inp_path = os.path.join(tempdir, f'{model.inp.name}.inp')

        # get a polygons dataframe and change it slightly
        polygons = model.inp.polygons
        model.inp.polygons = polygons.round(1)

        # save the updated model to disk
        model.inp.save(inp_path)

        # with this updated model, instantiate a new model object
        # and make changes to the Polygons once again
        model2 = swmmio.Model(inp_path)
        model2.inp.polygons = polygons.round(0)

        # save the model once more
        model2.inp.save()

        # confirm that there is only one instance of
        # the [Polygons] section in the inp file
        with open(inp_path, 'r') as file:
            data = file.read()
            assert data.count("[POLYGONS]") == 0
            assert data.count("[Polygons]") == 1
```

This test also uses an existing test inp file, this way we can omit the `swmmio/tests/data/example_polygons.inp` file from this PR.

**Review by aerispaha — CHANGES_REQUESTED** [review]
@BuczynskiRafal thanks for this great contribution to `swmmio`! Overall, your bug fix seems to work great. 

My only request is that we remove the new `swmmio/tests/data/example_polygons.inp` file from this PR, and use a test inp file that already exists in the repo. I'd also suggest we refactor the unit test as I described below for a couple reasons: 
1. this way test files aren't left behind if this test fails
2. I'd prefer to emphasize the `model.inp.<section> = <dataframe>` pattern for updating models, rather than using the `replace_inp_section` function directly.  

Let me know what you think!

**Comment by BuczynskiRafal** [comment]
@aerispaha Thanks for the good points. I've made the changes.

**Review by aerispaha — APPROVED** [review]
This looks great, @BuczynskiRafal! I'll get this integrated into the next release asap

**Pull request merged**